### PR TITLE
Note settings refactoring

### PR DESCRIPTION
**Pull request by kseniaSs**
Refactor noteSetting's services/repositories. 
Move functions from note.ts/note.service/note.repository.

### CODE REVIEW

File: src/application/services/useNoteSettings.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+   * NoteSettings ref
+   */
+  const noteSettings = ref<NotesSettings | null>(null);
+
+  /**
+   * Get note settings
+   *
+   * @param publicId - Note publicId
+   */
+  const loadSettings = async (publicId: string): Promise<void> => {
```
Comment (neSpecc) [inline src/application/services/useNoteSettings.ts:37]: it's better not to use the "public" prefix, because from the client-perspective there is only "id" property. It doesn't know about internal id which is a backend feature.
```suggestion
  const loadSettings = async (id: string): Promise<void> => {
```

File: src/infrastructure/storage/noteSettings.ts
Comment (neSpecc) [inline src/infrastructure/storage/noteSettings.ts:1]: I think we can get rid of Note Setting Store at the moment. We can add it later if we will need it.

File: src/application/services/useNoteSettings.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+   * NoteSettings ref
+   */
+  const noteSettings = ref<NotesSettings | null>(null);
+
+  /**
+   * Get note settings
+   *
+   * @param id - Note id
+   */
+  const loadSettings = async (id: string): Promise<void> => {
```
Comment (neSpecc) [inline src/application/services/useNoteSettings.ts:37]: it's better to use the `NoteId` type instead of `string`

File: src/domain/noteSettings.repository.interface.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+ */
+export default interface NoteSettingsRepositoryInterface {
+
+  /**
+   * Returns NotesSettings by publicId
+   *
+   * @param publicId - note publicId
+   * @returns NotesSettings | null - NotesSettings instance
+   */
+  getNotesSettingsById(publicId: string): Promise<NotesSettings | null>;
```
Comment (neSpecc) [inline src/domain/noteSettings.repository.interface.ts:14]: Here and following: publicId -> id

File: src/infrastructure/noteSettings.repository.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+   */
+  public async getNotesSettingsById(publicId: string): Promise<NotesSettings | null> {
+    const notesSettingsData = await this.noteStorage.getNotesSettingsById(publicId);
+
+    /**
+     * If notesSettings data in storage exists
+     */
+    if (notesSettingsData) {
+      return notesSettingsData;
+    }
```
Comment (neSpecc) [inline src/infrastructure/noteSettings.repository.ts:46]: these lines have no sense, since we have removed noteSettings storage

File: src/infrastructure/noteSettings.repository.ts [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+    /**
+     * If note data in API payload exists
+     */
+    if (notesSettings) {
+      /**
+       * Insert note to storage
+       */
+      await this.noteStorage.updateNotesSettings(notesSettings);
+    }
```
Comment (neSpecc) [inline src/infrastructure/noteSettings.repository.ts:61]: same

File: src/application/services/useNoteSettings.ts [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+   * @param id - Note id
+   */
+  const loadSettings = async (id: NoteId): Promise<void> => {
+    noteSettings.value = await noteSettingsService.getNotesSettingsById(id);
+  };
+
+
+  return {
+    noteSettings,
+    loadSettings,
```
Comment (neSpecc) [inline src/application/services/useNoteSettings.ts:45]: lets call them `settings` and `load`

File: src/domain/noteSettings.repository.interface.ts [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+ */
+export default interface NoteSettingsRepositoryInterface {
+
+  /**
+   * Returns NotesSettings by  id
+   *
+   * @param  id - note  id
+   * @returns NotesSettings | null - NotesSettings instance
+   */
+  getNotesSettingsById(id: string): Promise<NotesSettings | null>;
```
Comment (neSpecc) [inline src/domain/noteSettings.repository.interface.ts:14]:
```suggestion
  getNotesSettingsById(id: NoteId): Promise<NotesSettings | null>;
```

Comment (neSpecc) [reply src/domain/noteSettings.repository.interface.ts:14]: I think it should not return `null`. If an incorrect id will be passed, there will be 404 exception thrown

File: src/domain/noteSettings.service.ts [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+   */
+  constructor(noteSettingsRepository: NoteSettingsRepository) {
+    this.noteSettingsRepository = noteSettingsRepository;
+  }
+
+  /**
+   * Get notesSettings
+   *
+   * @param publicId - Note publicId
+   * @returns { NotesSettings | null } - Note data
```
Comment (neSpecc) [inline src/domain/noteSettings.service.ts:26]:
```suggestion
```

File: src/domain/noteSettings.service.ts [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+   * @param noteSettingsRepository - Note repository instance
+   */
+  constructor(noteSettingsRepository: NoteSettingsRepository) {
+    this.noteSettingsRepository = noteSettingsRepository;
+  }
+
+  /**
+   * Get notesSettings
+   *
+   * @param publicId - Note publicId
```
Comment (neSpecc) [inline src/domain/noteSettings.service.ts:25]: do not use the "public" prefix

File: src/infrastructure/noteSettings.repository.ts [hunk 10]
```diff
@@ -0,0 +1,66 @@
+import type NoteSettingsRepositoryInterface from '@/domain/noteSettings.repository.interface';
+import type NotesSettings from '@/domain/entities/NotesSettings';
+import type NoteStorage from '@/infrastructure/storage/note';
+import type NotesApiTransport from '@/infrastructure/transport/notes-api';
+import type { NoteId } from '@/domain/entities/Note';
+import type { GetNotesSettingsResponsePayload } from '@/infrastructure/transport/notes-api/types/GetNoteResponsePayload';
+
+/**
+ * Note repository
```
Comment (neSpecc) [inline src/infrastructure/noteSettings.repository.ts:9]:
```suggestion
 * Note settings repository
```

File: src/infrastructure/noteSettings.repository.ts (hunk 4 above)
Comment (neSpecc) [reply src/infrastructure/noteSettings.repository.ts]: Because there should be a clear reason to add a cache, and for now we don't have that reason.

File: src/application/services/useNoteSettings.ts [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+
+
+/**
+ * Application service for working with the Note settings
+ */
+export default function (): UseNoteSettingsComposableState {
+  /**
+   * NoteSettings ref
+   */
+  const note = ref<NotesSettings | null>(null);
```
Comment (slaveeks) [inline src/application/services/useNoteSettings.ts:31]: Why note?

File: src/domain/noteSettings.repository.interface.ts [hunk 12]
```diff
@@ -0,0 +1,17 @@
...
+export default interface NoteSettingsRepositoryInterface {
+
+  /**
+   * Returns NotesSettings by  id
+   *
+   * @param  id - note  id
+   * @throws Will throw an error if id is not found.
+   * @returns NotesSettings - NotesSettings instance
+   */
+  getNotesSettingsById(id: NoteId): Promise<NotesSettings>;
```
Comment (neSpecc) [inline src/domain/noteSettings.repository.interface.ts:16]:
```suggestion
  getNoteSettingsById(id: NoteId): Promise<NoteSettings>;
```

File: src/domain/noteSettings.repository.interface.ts [hunk 13]
```diff
@@ -0,0 +1,17 @@
+import type NotesSettings from '@/domain/entities/NotesSettings';
```
Comment (neSpecc) [inline src/domain/noteSettings.repository.interface.ts:1]:
```suggestion
import type NoteSettings from '@/domain/entities/NoteSettings';
```

File: src/domain/noteSettings.repository.interface.ts [hunk 14]
```diff
@@ -0,0 +1,17 @@
...
+import type { NoteId } from './entities/Note';
+
+/**
+ * Repository interface describes the methods that required by domain for its business logic implementation
+ */
+export default interface NoteSettingsRepositoryInterface {
+
+  /**
+   * Returns NotesSettings by  id
```
Comment (neSpecc) [inline src/domain/noteSettings.repository.interface.ts:10]:
```suggestion
   * Returns setting for a note with passed id
```

File: src/domain/noteSettings.service.ts [hunk 15]
```diff
@@ -0,0 +1,39 @@
...
+   * Note Service constructor
+   *
+   * @param noteSettingsRepository - Note repository instance
+   */
+  constructor(noteSettingsRepository: NoteSettingsRepository) {
+    this.noteSettingsRepository = noteSettingsRepository;
+  }
+
+  /**
+   * Get notesSettings
```
Comment (neSpecc) [inline src/domain/noteSettings.service.ts:24]:
```suggestion
   * Returns setting for a note with passed id
```

File: src/domain/noteSettings.service.ts [hunk 16]
```diff
@@ -0,0 +1,39 @@
...
+   *
+   * @param id - Note id
+   */
+  public async getNotesSettingsById(id: NoteId): Promise<NotesSettings | null> {
```
Comment (neSpecc) [inline src/domain/noteSettings.service.ts:28]:
```suggestion
  public async getNoteSettingsById(id: NoteId): Promise<NoteSettings | null> {
```

File: src/domain/noteSettings.service.ts [hunk 17]
```diff
@@ -0,0 +1,39 @@
...
+    let result;
+
+    try {
+      result = await this.noteSettingsRepository.getNotesSettingsById(id);
+    } catch (error) {
+      throw new Error(`404 Not Found: We couldn't find ${id} note.`);
+    }
```
Comment (neSpecc) [inline src/domain/noteSettings.service.ts:35]: I think, error handling and ui message should be at the presentation layer. Also you need to check the error for `NotFoundError` instance. 

Rough example:

```ts
/**
 * Get note settings
 *
 * @param id - Note id
 */
 const load = async (id: NoteId): Promise<void> => {
   try { 
     noteSettings.value = await noteSettingsService.getNotesSettingsById(id);
   } catch (error){
     if (error instanceof NotFoundError) {
       router.replace({
         name: "404"
       });
       
       return;
     }
     
     // other errors should be thrown down
     throw error;
   }
 };
```

I think error handling could be done via a separate PR.

File: src/infrastructure/noteSettings.repository.ts [hunk 18]
```diff
@@ -0,0 +1,42 @@
...
+    this.transport = notesApiTransport;
+  }
+
+  /**
+   * Get NotesSettings by note ID
+   *
+   * @param id - Note id
+   * @returns { NotesSettings } - NotesSettings instance
+   */
+  public async getNotesSettingsById(id: NoteId): Promise<NotesSettings> {
```
Comment (neSpecc) [inline src/infrastructure/noteSettings.repository.ts:39]:
```suggestion
  public async getNoteSettingsById(id: NoteId): Promise<NoteSettings> {
```

File: src/infrastructure/noteSettings.repository.ts [hunk 19]
```diff
@@ -0,0 +1,42 @@
...
+    return await this.transport.get<GetNotesSettingsResponsePayload>('/note/' + id + '/settings');
```
Comment (neSpecc) [inline src/infrastructure/noteSettings.repository.ts:40]:
```suggestion
    return await this.transport.get<GetNoteSettingsResponsePayload>('/note/' + id + '/settings');
```

File: src/infrastructure/noteSettings.repository.ts [hunk 20]
```diff
@@ -0,0 +1,42 @@
+import type NoteSettingsRepositoryInterface from '@/domain/noteSettings.repository.interface';
+import type NotesSettings from '@/domain/entities/NotesSettings';
+import type NoteStorage from '@/infrastructure/storage/note';
+import type NotesApiTransport from '@/infrastructure/transport/notes-api';
+import type { NoteId } from '@/domain/entities/Note';
+import type { GetNotesSettingsResponsePayload } from '@/infrastructure/transport/notes-api/types/GetNoteResponsePayload';
```
Comment (neSpecc) [inline src/infrastructure/noteSettings.repository.ts:6]:
```suggestion
import type { GetNoteSettingsResponsePayload } from '@/infrastructure/transport/notes-api/types/GetNoteResponsePayload';
```

Comment (neSpecc) [reply src/infrastructure/noteSettings.repository.ts:6]: probably, could be moved to the separate file

**Review by neSpecc — CHANGES_REQUESTED** [review]

File: src/infrastructure/noteSettings.repository.ts [hunk 21]
```diff
@@ -0,0 +1,42 @@
...
+export default class NoteSettingsRepository implements NoteSettingsRepositoryInterface {
+  /**
+   * Transport instance
+   */
+  private readonly transport: NotesApiTransport;
+
+  /**
+   * Note storage
+   */
+  private noteStorage: NoteStorage;
```
Comment (neSpecc) [inline src/infrastructure/noteSettings.repository.ts:20]: seems irrelevant

File: src/infrastructure/storage/note.ts [hunk 22]
```diff
@@ -41,24 +41,24 @@ export default class NoteStorage {
...
   /**
-   * Update notesSettings
+   * Update noteSettings
    *
-   * @param notesSettings - notesSettings to update
+   * @param noteSettings - noteSettings to update
    */
-  public async updateNotesSettings(notesSettings: NotesSettings): Promise<void> {
-    console.log('Update notesSettings', notesSettings);
+  public async updateNoteSettings(noteSettings: NoteSettings): Promise<void> {
```
Comment (neSpecc) [inline src/infrastructure/storage/note.ts:60]: unused code, better remove it

**Review by slaveeks — APPROVED** [review]

**Review by neSpecc — APPROVED** [review]
